### ISSUE TITLE
[feature request] cbv delta variant that takes an unfold-db

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#6144]
#### Version

8.7.0

#### Operating system

Linux

#### Description of the problem

I have a goal where `Hint Unfold <list> : foo. autounfold with foo.` is at least three orders of magnitude slower than `cbv [<list>]`.  Unfortunately, `cbv delta` lists are not extensible in the way that unfold databases are.  I'd like a way to use unfold databases that is not awfully slow, or a way to use `cbv` in an extensible manner.

**Comment by Zimmi48** [comment]
Would that be solved by having a way to iterate over a hint database in Ltac (as in https://stackoverflow.com/questions/42709641/is-is-possible-to-implement-a-coq-tactic-that-inspects-a-hintdb-if-so-how)?

**Comment by JasonGross** [comment]
No.  Well, maybe, but I'm dubious.  I'd need numerous calls to `constr_eq` (or Ltac's `context`) to check whether or not progress was made, because `cbv delta [foo bar]` will unfold both `foo` and `bar` regardless of whether or not one is defined in terms of the other (unlike `cbv delta [foo]; cbv delta [bar]`).  And I'd need to first issue all the `cbv delta` calls, followed by a single call to `cbv beta iota zeta`, and hope that the term I get by running a lot of `cbv delta` isn't too big.... The underlying issue here, in some sense, is that there's no way to build `cbv [foo bar]` from `cbv [foo]` and `cbv [bar]` without sacrificing a lot in performance.

**Comment by ppedrot** [comment]
@JasonGross Already accessible in the Ltac2 API btw.
